internal/service/apigateway/rest_api.go

```diff
@@ -308,12 +319,16 @@ func resourceRestAPICreate(d *schema.ResourceData, meta interface{}) error {
 			})
 		}
 
-		if v, ok := d.GetOk("policy"); ok && v.(string) != aws.StringValue(output.Policy) {
-			updateInput.PatchOperations = append(updateInput.PatchOperations, &apigateway.PatchOperation{
-				Op:    aws.String(apigateway.OpReplace),
-				Path:  aws.String("/policy"),
-				Value: aws.String(v.(string)),
-			})
+		if v, ok := d.GetOk("policy"); ok {
+			if equivalent, err := awspolicy.PoliciesAreEquivalent(v.(string), aws.StringValue(output.Policy)); err != nil || !equivalent {
```

Comment:
`err != nil || !equivalent` or `err != nil && !equivalent`?
If `err != nil` do we want to send the new value? The _validation covers error_ seems to suggest we don't.

Reply:
`PoliciesAreEquivalent()` is pretty sensitive. What normally happens is the `output.Policy` is nil, since it hasn't been set yet. `PoliciesAreEquivalent` then throws an error. I had more complicated logic so make all the scenarios easier to read but it boiled down to `err == nil && equivalent` means you _don't_ want to send the new value. Thus, `err != nil || !equivalent` should be the correct logic.